core/mutator/entry/entry.go

```diff
@@ -41,51 +42,75 @@ func New() *Entry {
 
 // Mutate verifies that this is a valid mutation for this item and applies
 // mutation to value.
-func (*Entry) Mutate(oldValue, mutation []byte) ([]byte, error) {
-	update := new(tpb.SignedKV)
-	if err := proto.Unmarshal(mutation, update); err != nil {
-		return nil, err
-	}
-
+func (*Entry) Mutate(oldValue, update proto.Message) ([]byte, error) {
 	// Ensure that the mutation size is within bounds.
 	if proto.Size(update) > mutator.MaxMutationSize {
 		glog.Warningf("mutation (%v bytes) is larger than the maximum accepted size (%v bytes).", proto.Size(update), mutator.MaxMutationSize)
 		return nil, mutator.ErrSize
 	}
 
+	updated, ok := update.(*tpb.SignedKV)
+	if !ok {
+		glog.Warning("received proto.Message is not of type *tpb.SignedKV.")
+		return nil, fmt.Errorf("updateM.(*tpb.SignedKV): _, %v", ok)
+	}
+	var oldEntry *tpb.Entry
+	if oldValue != nil {
+		oldEntry, ok = oldValue.(*tpb.Entry)
```

Comment:
`oldEntry` shadows the outer variable.  It's safer to declare a new variable name, and then set set `outer = inner`